bot/extensions/fun/weather.py

```diff
@@ -0,0 +1,129 @@
+from geopy.geocoders import Nominatim
+from timezonefinder import TimezoneFinder
+import pytz
+from pytz import country_timezones, timezone
+from datetime import datetime
+from discord.ext.commands import Cog, command
+import requests
+from discord import Embed
+import time
+import string
```

Comment:
You can optimize this import by doing `from string import capwords` since you're only using this function. Generally, it's a good idea to optimize your library once you know each function or part of this library you'll use.